### ISSUE TITLE
option to match a track as album

### ISSUE THREAD

**Issue by arogl** [snejus/beets-bandcamp#11]
Would it be possible to add an option for when an artists relaeases a track as a single to be treated as an album?

**Comment by snejus** [comment]
That's literally what I'm about to release in a couple of days. By default a singleton has 0 use with its limited metadata. I saw that beets 1.5.0 removes that limitation, so we can go ahead with it. For example, that's data I'm currently getting for a track that only exists on its own `bandcamp.com/track/track-name` without an associated album:

```yml
(new) album: Matriark - Arangel
(new) albumstatus: Official
(new) albumtype: single
artist: Matriark
(new) catalognum: ''
comments: "mt004\r\n\r\nmastering by alexander salomonsen\r\n\r\n\r\nI'm very happy\
  \ to announce Andrea's debut track. <3\r\n\r\nhttps://soundcloud.com/matriarkcph\r"
(new) country: SE
data_source: bandcamp
(new) day: 9
(adding soon)genre: ''
id: 1777
(new) label: Megatech
media: Digital Media
(new) month: 11
title: Arangel
(new) year: 2020
```

Basically, there used to be only `track_id`, `title`, `artist` and roughly `track_alt` available - you should see this when you try to import a singleton from musicbrainz - the output is ridiculously limited.

I just set the `albumtype` to `single` and ensure that both artist and the title are found in the album name, and the rest is generic metadata. I organise my folders by `label` at the top level, for example:

* Sinensis: two partial albums with a single track each
* Ute.Rec: three full albums. You can tell First Light is a compilation because tracks have artists names in them (and there's no artist name in the album folder name)
* UNNAMED (UNMD) artist has only released standalone tracks - so they live just below the top level. This is the case I refered to above. Since it's now possible to attach additional metadata, it's possible to organise them like this. It isn't though supported (unless I'm unaware) by the default musicbrainz autotagger :(
```
Sinensis
├── Alpha Tracks - SINE03
│   └── 02. Odondo.flac
└── Omformer - SINE04
    └── 02. Orbital.flac
Ute.Rec
├── First Light
│   ├── 01. Pyramid of Knowledge - Initiations.flac
│   ├── 02. Giorgo Celeste and The Cobra - The Other Side.flac
│   ├── 03. Ezy - Candy Flip.flac
│   ├── 04. CenterSonic - BumBum25 (Jaro Remake).flac
│   ├── 05. OT2 - O2matic.flac
│   ├── 06. Clementz - Fundament.flac
│   ├── 07. Rapid - Moonray (Tur Retur Frankfurt Mix).flac
│   ├── 08. Omformer - Feel.flac
│   ├── 09. Kineta - Cosmic Flux.flac
│   ├── 10. Mikkel Rev - Amneo.flac
│   ├── 11. Alpha Tracks - World Fair.flac
│   ├── 12. Mr. Free - Time 2 B Free.flac
│   ├── 13. HLLW - Elg I Soloppgang.flac
│   └── 14. Wilberforce - Unio Mystica.flac
├── Mikkel Rev - UTE004
│   ├── 01. The Human Experience (Empathy Mix).flac
│   ├── 02. Parallell.flac
│   ├── 03. Formulae.flac
│   └── 04. Biotope.flac
└── The Dosadi Experiments - UTE005
    ├── 01. The Hidden World.flac
    ├── 02. Arrival.flac
    ├── 03. Gowatchin Rituals.flac
    └── 04. Winds Of Tandaloor.flac
UNNAMED (UNMD)
├── UNNAMED (UNMD) - Jean-Kevin and La ThugLife.flac
├── UNNAMED (UNMD) - Le Monde Merveilleux des Gens Heureux.flac
└── UNNAMED (UNMD) - On l'appelait _Kevin and le grand Magicien_.flac

**Comment by snejus** [comment]
This has implemented in `0.7.0`. Will close this for now and will be looking forward to another ticket with any feedback or suggestions :)
